yarp/yarp_compiler.c

```diff
@@ -1185,11 +1186,12 @@ yp_compile_node(rb_iseq_t *iseq, const yp_node_t *node, LINK_ANCHOR *const ret,
           yp_statements_node_t *statements_node = (yp_statements_node_t *) node;
           yp_node_list_t node_list = statements_node->body;
           for (size_t index = 0; index < node_list.size; index++) {
-              // We only want to have popped == false for the last instruction
               if (!popped && (index != node_list.size - 1)) {
-                  popped = true;
+                  yp_compile_node(iseq, node_list.nodes[index], ret, src, Qtrue, compile_context);
```

Comment:
It seems like you're using `Qtrue` and `Qfalse` to represent `popped`, but that relies on the fact that `Qfalse` is 0. to convert into 0. Can we use a boolean instead?